### PR TITLE
Properties, Macros, Tests, Docs and smaller changes.

### PR DESCRIPTION
**Pull request by KDecay**
- Combined some macros to have a more flexible one. 
- Added some more border properties like `Color`,  `Style` and `Width` with the explicit and shorthand definitions.
- Added more tests.
- Added some docs.
- Smaller changes here and there.

### CODE REVIEW

**Review by geom3trik — COMMENTED** [review]
Currently some of the properties, like border for separate sides, are not supported by the renderer in vizia. Should we remove these or keep them for a time when a future vizia renderer might support them?

File: src/values/keywords/inset.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+use crate::{impl_parse, Parse};
+
+/// The 'inset' keyword.
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct InsetKeyword;
```
[the remark this answers is not in the document]
Comment (KDecay) [reply src/values/keywords/inset.rs:5]: The `inset` keyword is used for the `BoxShadow`. It changes the shadow from an outer shadow (outset) to an inner shadow (inset).

File: src/property.rs [hunk 2]
```diff
@@ -51,26 +53,45 @@ define_property! {
...
-        "border-width": BorderWidth(Length),
+        // ----- Border -----
 
         // Border Color
-        "border-color": BorderColor(Color),
+        "border-color": BorderColor(BorderColor),
+        "border-top-color": BorderTopColor(Color),
+        "border-right-color": BorderRightColor(Color),
+        "border-bottom-color": BorderBottomColor(Color),
+        "border-left-color": BorderLeftColor(Color),
```
[the remark this answers is not in the document]
Comment (KDecay) [reply src/property.rs]: Commented them out. See comment below.

File: src/values/border/border_style_keyword.rs [hunk 3]
```diff
@@ -0,0 +1,33 @@
+use crate::{define_enum, Parse};
+
+define_enum! {
+    /// The keyword of a single border style.
+    pub enum BorderStyleKeyword {
```
[the remark this answers is not in the document]
Comment (geom3trik) [reply src/values/border/border_style_keyword.rs:5]: Okay, that sounds good

**Pull request merged**